### ISSUE TITLE
 [Feature Request] Option To Save Parity Files To Separate Folder

### ISSUE THREAD

**Issue by jnite** [bobafetthotmail/par2drive#3]
For folders with many files, all the generated parity files tend to clutter up the source folder. It would be great if we could pass an optional folder path for saving all the parity files to. It would have an identical folder structure that mirrors the source path, but would only contain parity files. 

Or do you have any other suggestions for preventing source folders from being crowded with parity files?

**Comment by bobafetthotmail** [comment]
par2 has an argument to specify a different folder to store recovery files (the -B option) so it's easy to add the feature in my script.

I've uploaded a new version, tell me if it's ok.

**Comment by jnite** [comment]
Thanks for making this change. I also like the new ability to pass an option to par2. That was going to be my next feature request!

The changes work great when the path is a folder...but when the path is a single file I get an error. For example if I run this command...

`par2drive sync /home/me/sample.txt`

...I get this error:

```
Ignoring out of basepath source file: /home/me/sample.txt
You must specify a list of files when creating.
Warning! error detected, terminating batch processing.
```
Are you getting this error as well?

**Comment by bobafetthotmail** [comment]
Yeah, fixed that now.

I also split up par2 options in 3 different variables, one for each of its actions (create verify repair) as par2 will stop and error out if it finds an option it does not need (for example if you run a "par2 verify -r10 file" ).

I'm not 100% satisfied with specifying par2 options like that, but I didn't find simple or better ways given the limitation of shell script.

**Comment by jnite** [comment]
Thanks. 

This is interesting. Now single file paths are working again, but only for absolute paths. The error still occurs if you use a relative path, such as "./sample.txt".

As for the par2 options, I have an idea. Why not check for environment variables? For example, the script could check if a variable named PAR2DRIVE_CREATE_OPT is set, and if so use its value.

Or you could check for a configuration file, and read the option values from there, but that seems like a lot more work.

Oh, and one more thing: did you mean to default the create option's redundancy to 50%?

**Comment by bobafetthotmail** [comment]
> The error still occurs if you use a relative path, such as "./sample.txt".

Actually that's not because it's relative, I did test relative paths and they were working.

It breaks because it is a file in the same folder as the terminal (i.e. it is not ./some folder/sample.txt ), so it's kind of a corner case. Handled that now.

> As for the par2 options, I have an idea. Why not check for environment variables? 

Hm, that's good. I'll use a variation of that, thanks. 
options="-r10" par2drive sync "./Test Folder"

will now pass the -r10 to par2

> create option's redundancy to 50%?

No, it was left in from testing.

I've uploaded another version, should be ok now. Tell me how it goes.
